primitives/merkle-mountain-range/src/lib.rs

```diff
@@ -397,26 +397,46 @@ sp_api::decl_runtime_apis! {
 	/// API to interact with MMR pallet.
 	pub trait MmrApi<Hash: codec::Codec> {
 		/// Generate MMR proof for a leaf under given index.
-		fn generate_proof(leaf_index: LeafIndex) -> Result<(EncodableOpaqueLeaf, Proof<Hash>), Error>;
+		fn generate_proof(leaf_index: LeafIndex) -> Result<(EncodableOpaqueLeaf, BatchProof<Hash>), Error>;
```

Comment:
I don't think we should change the existing API.
Existing runtime API and RPC API should stay the same so we don't break existing users. Add the new "batch" functions.

```suggestion
		fn generate_proof(leaf_index: LeafIndex) -> Result<(EncodableOpaqueLeaf, Proof<Hash>), Error>;
```